plugins/core/hu.bme.mit.gamma.action.model/src/hu/bme/mit/gamma/action/util/ActionModelValidator.java

```diff
@@ -192,4 +193,25 @@ public ProcedureDeclaration getContainingProcedure(Branch branch) {
 		} 
 		throw new IllegalArgumentException("Unknown container for Branch.");
 	}
+	
+//////////////////////////////////////////////////////////////////////
+	public Collection<ValidationResultMessage> checkIfStatement(IfStatement ifs){
+		Collection<ValidationResultMessage> validationResultMessages = new ArrayList<ValidationResultMessage>();
+		// all Branch
+		for(Branch branch : ifs.getConditionals()) {
+			// Branch action is a Block
+			if(branch.getAction() instanceof Block) {
+				Block block = (Block) branch.getAction();
+				// Block is empty
+				if(block.getActions().isEmpty()) {
+					validationResultMessages.add(new ValidationResultMessage(ValidationResult.ERROR,
```

Comment:
Maybe this should be a warning, not an error. Also, why is this checked only in blocks contained by IfStatements? What about empty blocks e.g., in ProcedureDeclarations?